pkg/controller/infinispan/resources/config/xsite.go

```diff
@@ -87,10 +88,12 @@ func appendRemoteLocation(infinispan *ispnv1.Infinispan, remoteLocation *ispnv1.
 	return nil
 }
 
-func appendKubernetesRemoteLocation(infinispan *ispnv1.Infinispan, remoteLocation *ispnv1.InfinispanSiteLocationSpec,
-	remoteKubernetes *kube.Kubernetes, logger logr.Logger, xsite *config.XSite) error {
-	siteServiceName := infinispan.GetSiteServiceName()
-	namespacedName := types.NamespacedName{Name: siteServiceName, Namespace: infinispan.Namespace}
+func appendKubernetesRemoteLocation(infinispan *ispnv1.Infinispan, remoteLocation *ispnv1.InfinispanSiteLocationSpec, remoteKubernetes *kube.Kubernetes, logger logr.Logger, xsite *config.XSite) error {
+	name := xconditions.IfThenElse(remoteLocation.ClusterName == "", infinispan.Name, remoteLocation.ClusterName)
```

Comment:
Just my personal taste, but I would consider to use plain if/then/else construct: though it's more verbose I found it more readable and doesn't need dependencies.

Reply:
Sure, @rigazilla I would prefer to create our own function for this like `GetEnvWithDefault` rather than use third-party for it

Reply:
ok,
... ternary operator we miss you! :)

Reply:
Yeah, seems to never chance to get it in golang :)